File: medium/1262. Greatest Sum Divisible by Three/solution.py
class Solution:
    def maxSumDivThree(self, nums: List[int]) -> int:
        dp = [0, -10**9, -10**9]
        
        for num in nums:
            temp = dp[:]
            for i in range(3):
                remainder = (temp[i] + num) % 3
                dp[remainder] = max(dp[remainder], temp[i] + num)
        
        return dp[0]

File: medium/1262. Greatest Sum Divisible by Three/solution.rs
impl Solution {
    pub fn max_sum_div_three(nums: Vec<i32>) -> i32 {
        let mut dp = vec![0, i32::MIN, i32::MIN];
        
        for num in nums {
            let temp = dp.clone();
            for i in 0..3 {
                if temp[i] != i32::MIN {
                    let sum = temp[i] + num;
                    let remainder = (sum % 3) as usize;
                    dp[remainder] = dp[remainder].max(sum);
                }
            }
        }
        
        dp[0]
    }
}